### ISSUE TITLE
My VS Code lags when I enable Wakatime 

### ISSUE THREAD

**Issue by AF1QUE** [wakatime/vscode-wakatime#145]
Hello, I really like the addon - WakaTime but it lags my coding experience. When I start typing for a while my text appears a little bit late and lags a lot. I have tested the visual studio code by disabling your addon and it worked fine, it only lags when I use Wakatime. 

FYI, my specs are -
Intel Core i5 6th gen
8gb ddr3 ram
Windows 10

**Comment by igorskyflyer** [comment]
+1

Lags with the default config and I found in some of the comments, there is an option to enable/disable standalone mode - which should make it faster, same issue. Currently the WakaTime Python script keeps my CPU very busy when coding.

Thank you.

**Comment by verbruggenalex** [comment]
Yeah, I was having some issues with CPU as well, using Theia IDE (VS code in browser) in a docker environment. It kept going over 100% CPU and caused lagging in the typing in the IDE.

The next day I got an email from Wakatime saying: "You have many duplicate heartbeat events."

```
Hey there,

We noticed some unusual coding activity from your VS Code WakaTime plugin Yesterday:
You have many duplicate heartbeat events.

In rare cases, other VS Code extensions can cause problems with WakaTime. Please make sure your other VS Code extensions are not simulating cursor movements or saving files in the background. This can trick the WakaTime plugin into thinking you were typing even when you weren’t using your VS Code editor.

Most likely you’ve been bitten by a bug in the WakaTime plugin. To help us out, please open a new issue on our GitHub issue tracker.
```

The problem now seems to have disappeared. I'm not sure what caused it. I'll just leave the details of my docker image here should anyone want to check it out for debugging.

```
git clone git@github.com:verbruggenalex/docker-theia-php.git
cd docker-theia-php
docker-compose up -d
visit IDE at port 3000 in your browser
```

**Comment by alanhamlett** [comment]
I'm not able to reproduce this. Maybe it depends on the contents of your `index.php` file?

**Comment by alanhamlett** [comment]
It might be that extension, but not limited to ValeryanM.vscode-phpsab plugin, since another person having this issue doesn't have that plugin installed:
https://github.com/wakatime/vscode-wakatime/issues/163#issuecomment-748996767

**Comment by alanhamlett** [comment]
Might be fixed in v7.0.0 of the vscode extension, now that we're using the new [wakatime-cli](https://github.com/wakatime/wakatime-cli) that no longer needs Python.
